File: 278-first-bad-version/278-first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        left = 1
        right = n
        while(left<right):
            tmp = int((left+right)/2)
            if(isBadVersion(tmp)): 
                right = tmp 
            else: left = tmp+1
        return right

File: 278-first-bad-version/278-first-bad-version.js
/**
 * Definition for isBadVersion()
 * 
 * @param {integer} version number
 * @return {boolean} whether the version is bad
 * isBadVersion = function(version) {
 *     ...
 * };
 */

/**
 * @param {function} isBadVersion()
 * @return {function}
 */
var solution = function(isBadVersion) {
    /**
     * @param {integer} n Total versions
     * @return {integer} The first bad version
     */
    return function(n) {
        left = 1
        right = n
        while(left<right) {
            tmp = parseInt((left+right)/2)
            if(isBadVersion(tmp)) right = tmp 
            else left = tmp+1
        }
        return right
    };
};

        